src/renderer/views/Watch/Watch.vue

```diff
@@ -42,7 +42,38 @@
           @toggle-theatre-mode="useTheatreMode = !useTheatreMode"
         />
         <div
-          v-if="!isLoading && (isUpcoming || errorMessage)"
+          v-if="!isLoading && isUpcoming && playabilityStatus === 'OK'"
+          class="trailer premiereDate"
+        >
+          <font-awesome-icon
+            :icon="['fas', 'satellite-dish']"
+            class="premiereIcon"
+          />
+          <p
+            v-if="upcomingTimestamp !== null"
+            class="premiereText"
+          >
+            <span
+              class="premiereTextTimeLeft"
+            >
+              {{ $t("Video.Premieres") }} {{ upcomingTimeLeft }}
+            </span>
+            <br>
+            <span
+              class="premiereTextTimestamp"
+            >
+              {{ upcomingTimestamp }}
+            </span>
+          </p>
+          <p
+            v-else
+            class="premiereText"
+          >
+            {{ $t("Video.Starting soon, please refresh the page to check again") }}
+          </p>
+        </div>
+        <div
+          v-else-if="!isLoading && (isUpcoming || errorMessage)"
```

Comment:
I'm on mobile so I might be missing something glaringly obvious but this looks like it mostly duplicates upcoming video the code below it.

Reply:
It's slightly different. It moves the premiere timestamp below the video and gets rid of the video thumbnail. I'll see if i can use some css classes and v-if statements to avoid the code duplication